src/viking.js

```diff
@@ -1,11 +1,182 @@
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(damage) {
+    super.receiveDamage(damage);
+    if (this.health > 0) {
+        // console for trial game
+      console.log(`${this.name} has received ${damage} points of damage`);
+      return `${this.name} has received ${damage} points of damage`;
+    } else {
+      console.log(`${this.name} has died in act of combat`);
+      return `${this.name} has died in act of combat`;
+    }
+  }
+  battleCry() {
+    //console for trial game
+    console.log("Odin Owns You All!");
+    return "Odin Owns You All!";
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  receiveDamage(damage) {
+    super.receiveDamage(damage);
+    if (this.health > 0) {
+        // console for trial game
+      console.log(`A Saxon has received ${damage} points of damage`);
+      return `A Saxon has received ${damage} points of damage`;
+    } else {
+      console.log(`A Saxon has died in combat`);
+      return `A Saxon has died in combat`;
+    }
+  }
+}
 
 // War
-class War {}
+class War {
+  addViking(vikingObj) {
+    this.vikingArmy.push(vikingObj);
+  }
+
+  addSaxon(saxonObj) {
+    this.saxonArmy.push(saxonObj);
+  }
+
+  vikingAttack() {
+    if (this.vikingArmy.length === 0) {
+      this.showStatus();
+      return;
+    }
+    let randomSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+    let randomViking = Math.floor(Math.random() * this.vikingArmy.length);
+    let attackResult = this.saxonArmy[randomSaxon].receiveDamage(
+      this.vikingArmy[randomViking].attack()
+    );
+    if (this.saxonArmy[randomSaxon].health <= 0) {
+      let index = this.saxonArmy[randomSaxon];
+      this.saxonArmy.splice(index, 1);
+    }
+    return attackResult;
+  }
+
+  saxonAttack() {
+    if (this.saxonArmy.length === 0) {
+      this.showStatus();
+      return;
+    }
+    let randomSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+    let randomViking = Math.floor(Math.random() * this.vikingArmy.length);
+    let attackResult = this.vikingArmy[randomViking].receiveDamage(
+      this.saxonArmy[randomSaxon].attack()
+    );
+    if (this.vikingArmy[randomViking].health <= 0) {
+      let index = this.vikingArmy[randomViking];
+      this.vikingArmy.splice(index, 1);
+    }
+    return attackResult;
+  }
+
+  showStatus() {
+    if (this.vikingArmy.length !== 0 && this.saxonArmy.length !== 0) {
+        return "Vikings and Saxons are still in the thick of battle.";
+      }
+    else if (this.saxonArmy.length === 0) {
+      return "Vikings have won the war of the century!";
+    }
+    else if (this.vikingArmy.length === 0) {
+      return "Saxons have fought for their lives and survived another day...";
+    }
+  }
+  constructor() {
```

Comment:
Haha yes, I think I did it because I was paying too much attention to the order in which it was explained in the instructions, but I can see how it makes sense to put it at the beginning